CMakeLists.txt

```diff
@@ -160,5 +143,15 @@ foreach(MODULE ${THIRD_PARTY_MODULES})
   endif()
 endforeach()
 
+if(NOT PCRE_LIBRARY)
+  target_include_directories(pcre PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/pcre")
+  target_compile_definitions(pcre PUBLIC "PCRE_STATIC=1")
+endif()
+
 set(THIRD_PARTY_INCLUDE_PATHS ${EXTRA_INCLUDE_PATHS} CACHE INTERNAL "" FORCE)
-set(THIRD_PARTY_DEFINITIONS ${EXTRA_DEFINES} CACHE INTERNAL "" FORCE)
+
+# boost checks
+find_package(Boost 1.51.0 COMPONENTS context filesystem program_options regex system thread REQUIRED)
+add_library(boost INTERFACE)
```

Comment:
If we require CMake 3+ you don't need this and can instead use `Boost::context`, `Boost::filesystem` `Boost::program_options`, `Boost::regex`, `Boost::system` and `Boost::thread` directly as dependencies after calling `find_package`. (it also means you don't need the `target_include_directories` or `target_link_libraries` either)

I don't know what version of CMake is on our supported platforms though :(